### PR TITLE
AST-449 - improvement: Added heading font-weight support to widget title & added all font props support in GB editor as well

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- This PRs covered following 2 improvements
- 1. Font weight not applying to widget title font
- 2. Customizer fonts not reflecting in GB editor
- Managed backwards to prevent direct change on existing users site

### Screenshots
- 1. Font weight issue 
- https://share.getcloudapp.com/kpuKedD6
- https://share.getcloudapp.com/Blu4qEGD
- 2. Customizer fonts not reflecting in GB editor
- Till master view - https://share.bsf.io/xQu74Qpd (all fonts, weights applied from customizer)
- With this PR - https://share.bsf.io/4gun0Wjl (all fonts, weights applied from customizer)
- 3. Same for colors as well - https://share.bsf.io/JruDj5Dr (all colors are applied from customizer)
- Till master view - https://share.bsf.io/xQu74Qpd 
- With this PR - https://share.bsf.io/04u2GR2E

### Types of changes
- Bug fix 
- Improvement

### How has this been tested?
- Cases has been added here - https://wp-astra.atlassian.net/browse/AST-449
- Review these with following filters -
- For enabling support
- `add_filter( 'astra_heading_fonts_typo_support', '__return_true' );`
- For updating HTML tag for widget title
- https://wpastra.com/docs/change-sidebar-widget-title-heading-tag/
- https://wpastra.com/docs/change-footer-widget-title-heading-tag/

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/theme-update/class-astra-theme-background-updater.php [hunk 1]
```diff
@@ -83,6 +83,7 @@ class Astra_Theme_Background_Updater {
 			),
 			'3.5.0' => array(
 				'astra_update_related_posts_grid_layout',
+				'astra_headings_font_support',
```
Comment (premanshup) [inline inc/theme-update/class-astra-theme-background-updater.php:86]: @imnavanath this should be added in v3.5.1.

File: inc/addons/heading-colors/assets/js/unminified/customizer-preview.js [hunk 2]
```diff
@@ -15,21 +15,30 @@
 	 */
 	astra_css( 'astra-settings[heading-base-color]', 'color', 'h1, .entry-content h1, h2, .entry-content h2, h3, .entry-content h3, h4, .entry-content h4, h5, .entry-content h5, h6, .entry-content h6' );
 
+	var h1_widget_title = '',
+		h2_widget_title = '',
+		h3_widget_title = '';
```
Comment (sagarb90) [inline inc/addons/heading-colors/assets/js/unminified/customizer-preview.js:20]: Can you write optimize way like this `var h1_widget_title, h2_widget_title, h3_widget_title = '';`

Comment (imnavanath) [reply inc/addons/heading-colors/assets/js/unminified/customizer-preview.js]: Done 👍

File: inc/theme-update/class-astra-theme-background-updater.php (hunk 1 above)
Comment (imnavanath) [reply inc/theme-update/class-astra-theme-background-updater.php:88]: Done 👍

**Review by abhishek-bsf — APPROVED** [review]

**Pull request merged**